lib/css_parser/rule_set.rb

```diff
@@ -237,9 +243,38 @@ def normalize_property(property)
     alias []= add_declaration!
     alias remove_declaration! delete
 
-    def initialize(selectors, block, specificity = nil)
+    def initialize(*args, selectors: nil, block: nil, offset: nil, filename: nil, specificity: nil) # rubocop:disable Metrics/ParameterLists
+      if args.any?
+        if selectors || block || offset || filename || specificity
+          raise ArgumentError, "don't mix positional and keyword arguments"
+        end
+
+        warn '[DEPRECATION] positional arguments are deprecated use keyword instead.', uplevel: 1
+
+        case args.length
+        when 2
+          selectors, block = args
+        when 3
+          selectors, block, specificity = args
+        when 4
+          filename, offset, selectors, block = args
+        when 5
+          filename, offset, selectors, block, specificity = args
+        else
+          raise ArgumentError
+        end
+      end
+
       @selectors = []
       @specificity = specificity
+
+      unless offset.nil? == filename.nil?
+        raise ArgumentError, 'require both offset and filename'
```

Comment:
code says something different:
```suggestion
        raise ArgumentError, 'require both offset and filename or no offset and no filename'
```